src/CentrifugeRouter.sol

```diff
@@ -32,14 +35,15 @@ contract CentrifugeRouter is Auth, ICentrifugeRouter {
     }
 
     modifier protected() {
-        if (_initiator == UNSET_INITIATOR) {
+        address currentInitiator = _initiator();
+        if (currentInitiator == address(0)) {
             // Single call re-entrancy lock
-            _initiator = msg.sender;
+            INITIATOR_SLOT.tstore(msg.sender);
             _;
-            _initiator = UNSET_INITIATOR;
+            INITIATOR_SLOT.tstore(address(0));
```

Comment:
Not sure if we need this line which is 100 gas more.
After each transaction the values in tstorage are discarded. The wyay I understand it is that the value are back to default ones

Reply:
True, should not be needed now that we are using transient storage. Good find

Reply:
@peculiarity a lot of tests are failing if I remove this. I guess the test case is 1 transaction, and if you do multiple calls separately to the router within 1 test, you need the reset above.

I guess this is also the reason you need it anyways: if some contract wants to call the router several times in 1 method (although it does not make much sense because you have multicall), you need the same.

So I would keep it then

Reply:
Yeah, I wasn't really sure. Let's leave it then